serpextract/serpextract.py

```diff
@@ -506,23 +490,26 @@ def get_all_query_params():
 
     return list(all_params)
 
-
+qs_params = defaultdict(list)
 def get_all_query_params_by_domain():
     """
     Return all the possible query string params for all search engines.
 
     :returns: a ``list`` of all the unique query string parameters that are
               used across the search engine definitions.
     """
+    global qs_params
+    if qs_params:
+        return qs_params
     engines = _get_search_engines()
     param_dict = defaultdict(list)
     for domain, parser in iteritems(engines):
-        # Find non-regex params
         params = {param for param in parser.keyword_extractor
                   if isinstance(param, string_types)}
-        tld_res = tldextract.extract(domain)
```

Comment:
Nitpick: Since you didn't actually change anything here, please change the name back to what it was and add the comment back.